tasks/build/preconfigure-k3s-auto-deploying-manifests.yml

```diff
@@ -12,7 +12,7 @@
 - name: Ensure Auto-Deploying Manifests are copied to controllers
   ansible.builtin.template:
     src: "{{ item }}"
-    dest: "{{ k3s_server_manifests_dir }}/{{ item | basename | replace('.j2','') }}"
+    dest: "{{ k3s_server_manifests_dir }}/{{ item | basename | replace('.j2','.yaml') }}"
```

Comment:
Why not name your template `manifest.yaml.j2` ?

Reply:
Hi

Here's the original PR that introduced this feature: https://github.com/PyratLabs/ansible-role-k3s/pull/5#issue-361736084

The playbook in use is as follows, it includes the extension before the `.j2` extension. I believe this is fairly common practice as it helps to identify the file format within the jinja2 template.

```yaml
- include_role:
    name: xanmanning.k3s
  vars:
    k3s_release_version: v1.17.0+k3s.1
    k3s_write_kubeconfig_mode: 644  
    k3s_server_manifests_templates:
      - "ns-monitoring.yaml.j2"
      - "ns-operators.yaml.j2"
```

The current behavior seems more flexible:

```text
ns-monitoring.yaml.j2 -> ns-monitoring.yaml
ns-monitoring.j2.yaml -> ns-monitoring.yaml
ns-monitoring.j2 -> ns-monitoring  # Not recognized by k3s
```

The change in your PR would cause the manifests to behave as below, which would allow all to be picked up, however becomes a bit messy in terms of file extensions:

```text
ns-monitoring.yaml.j2 -> ns-monitoring.yaml.yaml
ns-monitoring.j2.yaml -> ns-monitoring.yaml.yaml
ns-monitoring.j2 -> ns-monitoring.yaml
```